File: app/src/main/java/beans/prj/newsapp/NewsApplication.java
package beans.prj.newsapp;

import dagger.android.DaggerApplication;
import dagger.android.AndroidInjector;
import beans.prj.newsapp.di.DaggerAppComponent;

public class NewsApplication extends DaggerApplication {

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder().create(this);
    }

}

File: app/src/main/java/beans/prj/newsapp/NewsApplication.kt
package beans.prj.newsapp

import beans.prj.newsapp.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication

class NewsApplication : DaggerApplication() {

        override fun applicationInjector(): AndroidInjector<out DaggerApplication>
                = DaggerAppComponent.factory().create(this)
}